gobblin-runtime/src/main/java/org/apache/gobblin/runtime/mapreduce/GobblinWorkUnitsInputFormat.java

```diff
@@ -89,9 +100,13 @@ public static int getMaxMapper(Configuration conf) {
       if (inputs == null) {
         throw new IOException(String.format("Path %s does not exist.", path));
       }
+
       log.info(String.format("Found %d input files at %s: %s", inputs.length, path, Arrays.toString(inputs)));
+
       for (FileStatus input : inputs) {
         allPaths.add(input.getPath().toString());
+        blockLocationNamesForWorkUnitPaths.put(input.getPath().toString(),
```

Comment:
Can we not just get the block locations from the FS API here instead of from the de-serialized workunit file?

Reply:
So I was considering doing it here, but the first thing is that we want the block locations for the files specified by the work unit, not the work unit file itself. Because of this, we still have to deserialize the work unit file, and then the issue was that I don't think the actual file path is listed in the work unit, only the dataset pattern - the actual file paths for work units are only retrieved in the CopySource ...

Reply:
Although now that I think of it, I think it must be in the WorkUnit somewhere so that the extractor is able to open an inputstream to it ... I will look into it further to see if it can be done here, since I agree that it would be a better, more general solution

Reply:
Ok, so we have to be able to deserialize the CopyableFile/CopyEntity from the WorkUnit in the GobblinWorkUnitsInputFormat class, but the problem is that the CopyableFile/CopyEntity classes are in the gobblin-data-management module which I don't think should be a dependency of gobblin-runtime ... Also, the split offset info is json serialized from a DistcpFileSplitter.Split object, which is also in the gobblin-data-management module.